staging/src/k8s.io/apiserver/pkg/util/feature/feature_gate.go

```diff
@@ -73,7 +108,8 @@ const (
 // a string like feature1=true,feature2=false,...
 type FeatureGate interface {
 	// AddFlag adds a flag for setting global feature gates to the specified FlagSet.
-	AddFlag(fs *pflag.FlagSet)
+	// If a component is specified, only feature applies to those components will be displayed in flag help text.
+	AddFlag(fs *pflag.FlagSet, component string)
```

Comment:
since we're effectively using Enums here, how about defining a new `type Component string` alias, then specifying it here? Prevents people getting it wrong then

Reply:
added Component type.